pkg/smartcontract/rpcbinding/binding.go

```diff
@@ -413,6 +423,10 @@ func Generate(cfg binding.Config) error {
 				mfst.ABI.Methods = dropStdMethods(mfst.ABI.Methods, standard.Nep11NonDivisible)
 				ctr.IsNep11ND = true
 			}
+			if standard.ComplyABI(cfg.Manifest, standard.Nep24Royalty) == nil {
+				mfst.ABI.Methods = dropStdMethods(mfst.ABI.Methods, standard.Nep24Royalty)
+				ctr.IsNep24 = true
+			}
```

Comment:
And this way it will never work because you have `dropStdMethods` calls above for NEP-11.

Reply:
OK, we need a refactor of this code. Move `dropStdMethods` calls out of cycle. And then we may have a fully-qualified `switch std` with three options: NEP-11, NEP-17 and NEP-24. And be careful with `break // Can't be NEP-17 at the same time.`